satpy/tests/writer_tests/test_cf.py

```diff
@@ -293,6 +293,24 @@ def test_single_time_value(self):
                 bounds_exp = np.array([[start_time, end_time]], dtype='datetime64[m]')
                 np.testing.assert_array_equal(f['time_bnds'], bounds_exp)
 
+    def test_time_coordinate_on_a_swath(self):
+        """Test that time dimension is not added on swath data with time already as a coordinate."""
+        import xarray as xr
+
+        from satpy import Scene
+        scn = Scene()
+        test_array = np.array([[1, 2], [3, 4], [5, 6], [7, 8]])
+        scn['test-array'] = xr.DataArray(test_array,
+                                         dims=['y', 'x'],
+                                         coords={'time': ('y', np.array(['2018-05-30T10:05:00',
+                                                                         '2018-05-30T10:05:01',
+                                                                         '2018-05-30T10:05:02',
+                                                                         '2018-05-30T10:05:03'], dtype=np.datetime64))})
+        with TempFile() as filename:
+            scn.save_datasets(filename=filename, writer='cf')
```

Comment:
With `save_datasets(..., pretty=True)` the writer should not prefix the time coordinate with the dataset name, see https://github.com/pytroll/satpy/blob/3e94d59bad6f89306b19306f2e46b73df0e21d7b/satpy/writers/cf_writer.py#L37

Reply:
Thanks, I'll use this. The weird thing is that in the previously existing test `test_single_time_value()` this isn't needed.

Reply:
That is indeed strange... I'll add that to my list, will probably come up again when I'm gonna work on #1686